joblib/test/test_parallel.py

```diff
@@ -1260,13 +1260,23 @@ def test_memmapping_leaks(backend, tmpdir):
 
     # Make sure that the shared memory is cleaned at the end when we exit
     # the context
-    assert not os.listdir(tmpdir)
+    for _ in range(100):
```

Comment:
This test failed for no apparent reason at the last commit. I am tempted to re-write it as an isolated subprocess script to limit flakiness and better debug it. WDYT @ogrisel ?

Reply:
As you wish. I am not sure how it's going to help make it easier to debug but feel free to do so.

Reply:
I cannot reproduce the error that happened last time so let's leave this as is for now.